### PR TITLE
Fix pipeline predict transform methods

### LINKED ISSUE #19289: Call to pipeline.predict() fails
### System Info

- `transformers` version: 4.21.1
- Platform: macOS-12.5.1-arm64-arm-64bit
- Python version: 3.9.12
- Huggingface_hub version: 0.2.1
- PyTorch version (GPU?): 1.12.1 (False)
- Tensorflow version (GPU?): 2.9.2 (False)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: No

### Who can help?

@narsil

### Information

- [ ] The official example scripts
- [X] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [ ] My own task or dataset (give details below)

### Reproduction

Execute the following piece of code resulted in an exception that is pasted below.

```python
from transformers import pipeline

pipe = pipeline("text-classification")
print(pipe.predict(["This restaurant is awesome"]))

```

Exception:

```
Traceback (most recent call last):
  File "pipeline_test.py", line 5, in <module>
    print(pipe.predict(["This restaurant is awesome"]))
  File "miniconda3/envs/mlflow-py3.9/lib/python3.9/site-packages/transformers/pipelines/base.py", line 840, in predict
    return self(X=X)
  File "miniconda3/envs/mlflow-py3.9/lib/python3.9/site-packages/transformers/pipelines/text_classification.py", line 138, in __call__
    result = super().__call__(*args, **kwargs)
TypeError: __call__() missing 1 required positional argument: 'inputs'
```

### Expected behavior

Successful predictions as shown below

```
[{'label': 'POSITIVE', 'score': 0.9998743534088135}]
```

### Proposed fix

I dig a bit deeper into the implementation based on the exception and found out that this [change](https://github.com/huggingface/transformers/compare/main...s-udhaya:transformers:fix_pipeline_predict#diff-441f558737166b045444da9c4be81f566b3d69054e8f20e288aed746a691fa61R845) fixes the issue. If this indeed a fix, I am happy to create a PR.

### PR DESCRIPTION
**Pull request by s-udhaya**
# What does this PR do?

This PR fixes pipeline's predict and transform methods which are wrapper around ` __call__` of pipeline class. `__call__ `requires one positional argument, However these wrapper methods pass the incoming argument as a keyword argument to `__call__` method which leads to failure. Hence in this commit the keyword argument is modified to positional argument and basic tests are added to make sure this does not break again in future.

Fixes #19289

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@LysandreJik @Narsil

### CODE REVIEW

File: tests/pipelines/test_pipelines_common.py [hunk 1]
```diff
@@ -423,6 +425,33 @@ def test_unbatch_attentions_hidden_states(self):
         self.assertEqual(len(outputs), 20)
 
 
+@require_torch_and_tf
```
Comment (Narsil) [inline tests/pipelines/test_pipelines_common.py:428]: We cannot allow this, as the tests are run independantly, no test will ever by ran with the two frameworks !


You tests are good, just:
- Remove `parameterized` use two different tests each time, it's much easier  to read and parse (and execute separately).
- Remove `require_torch_and_tf`.

**Review by Narsil — COMMENTED** [review]
Thanks for this PR !

I think the fix is good, as I've mentioned in the issue, it might break something else since it's super old code, but I'm not sure.
Pinging @mfuntowicz that might know more.

Also I suggested edits to make the tests better.
I can take care of that if you want.

@sgugger too for another eye.

File: tests/pipelines/test_pipelines_common.py (hunk 1 above)
Comment (s-udhaya) [reply tests/pipelines/test_pipelines_common.py]: @Narsil Many thanks for the review and the suggestion. Updated the PR.

File: tests/pipelines/test_pipelines_common.py [hunk 2]
```diff
@@ -423,6 +423,53 @@ def test_unbatch_attentions_hidden_states(self):
         self.assertEqual(len(outputs), 20)
 
 
+class PipelineScikitCompatTest(unittest.TestCase):
+
+    def test_pipeline_predict_pt(self):
```
Comment (Narsil) [inline tests/pipelines/test_pipelines_common.py:428]: You do need to protect them with `@require_torch` and `@require_tf` though.

Comment (s-udhaya) [reply tests/pipelines/test_pipelines_common.py:429]: ;) yea right. Done .

**Review by sgugger — APPROVED** [review]
Thanks for the fix!

**Pull request merged**